sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py

```diff
@@ -0,0 +1,2472 @@
+                    "azure.identity package not installed. Please install it using 'pip install azure-identity'"
+                )
+            client = AsyncAzureOpenAI(
+                # See https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity?view=azure-python#azure-identity-get-bearer-token-provider
+                azure_ad_token_provider=get_bearer_token_provider(
+                    connection.token_credential, "https://cognitiveservices.azure.com/.default"
+                ),
+                azure_endpoint=connection.endpoint_url,
+                api_version=api_version,
+            )
+        elif connection.authentication_type == AuthenticationType.SAS:
+            logger.debug("[InferenceOperations.get_azure_openai_client] Creating AzureOpenAI using SAS authentication")
+            client = AsyncAzureOpenAI(
+                azure_ad_token_provider=get_bearer_token_provider(
+                    connection.token_credential, "https://cognitiveservices.azure.com/.default"
+                ),
+                azure_endpoint=connection.endpoint_url,
+                api_version=api_version,
+            )
+        else:
+            raise ValueError("Unknown authentication type")
+
+        return client
+
+
+class ConnectionsOperations(ConnectionsOperationsGenerated):
+
+    @distributed_trace_async
+    async def get_default(
+        self, *, connection_type: ConnectionType, with_credentials: bool = False, **kwargs: Any
+    ) -> ConnectionProperties:
+        """Get the properties of the default connection of a certain connection type, with or without
+        populating authentication credentials.
+
+        :param connection_type: The connection type. Required.
+        :type connection_type: ~azure.ai.projects.models._models.ConnectionType
+        :param with_credentials: Whether to populate the connection properties with authentication credentials. Optional.
+        :type with_credentials: bool
+        :return: The connection properties, or `None` if there are no connections of the specified type.
+        :rtype: ~azure.ai.projects.models._models.ConnectionProperties
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        kwargs.setdefault("merge_span", True)
+        if not connection_type:
+            raise ValueError("You must specify an connection type")
+        # Since there is no notion of default connection at the moment, list all connections in the category
+        # and return the first one
+        connection_properties_list = await self.list(connection_type=connection_type, **kwargs)
+        if len(connection_properties_list) > 0:
+            if with_credentials:
+                return await self.get(
+                    connection_name=connection_properties_list[0].name, with_credentials=with_credentials, **kwargs
+                )
+            else:
+                return connection_properties_list[0]
+        else:
+            return None
```

Comment:
returning None is not very user-friendly... would it make sense to return an empty ConnectionProperties or raise a ResourceNotFound error?

Reply:
Thanks @kristapratico, I updated the code to throw and exception instead:
` raise ResourceNotFoundError(f"No connection of type {connection_type} found")`